mlflow/models/evaluation/default_evaluator.py

```diff
@@ -636,9 +640,10 @@ def _log_model_explainability(self):
             _logger.debug("", exc_info=True)
             return
         try:
-            mlflow.shap.log_explainer(
-                explainer, artifact_path=_gen_log_key("explainer", self.dataset_name)
-            )
+            if not self.is_baseline_model:
```

Comment:
Our model validation only depends on scalar metrics, right? If so, I don't think we should call `_log_model_explainability()` at all if `is_baseline_model` is True.

In general, I think that adding all of these checks for `is_baseline_model` to determine whether or not we log to MLflow Tracking is somewhat brittle / error-prone. Instead, can we split `_evaluate()` into multiple methods, such as `_get_model_predictions()`, `_compute_metrics()` (with an argument for `log_to_mlflow_tracking=True/False`), and `_compute_explanations()`. For the candidate model, we'll call all 3 methods and pass `log_to_mlflow_tracking=True` to `_compute_metrics()`. For the baseline model, we'll only call `_get_model_predictions()` and `_compute_metrics()` with `log_to_mlflow_tracking=False`.

Let me know what you think here.

Reply:
Agree! I refactored `DefaultEvaluator.evalaute` to multiple methods: `_generate_model_predictions`, `_compute_metrics`, `_evaluate_custom_metrics_and_log_produced_artifacts` and `_log_metrics_and_artifacts`.